### PR TITLE
[controller] Fix a bug in deleteOneStoreVersion

### PR DESCRIPTION
**Pull request by lluwm**
In VHA::deleteOneStoreVersion, there is a bug that trying to get the push status after the push monitor is stopped. It is introduced in PR d7997d782. This PR fixes the issue by calling hasFatalDataValidationError before stopMonitorOfflinePush.

## How was this PR tested?
Passed CI

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Review by majisourav99 — APPROVED** [review]
LGTM

File: services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java [hunk 1]
```diff
@@ -3263,16 +3263,19 @@ private void deleteOneStoreVersion(String clusterName, String storeName, int ver
       deleteHelixResource(clusterName, resourceName);
       LOGGER.info("Killing offline push for: {} in cluster: {}", resourceName, clusterName);
       killOfflinePush(clusterName, resourceName, true);
+
+      // Check DIV error in the push status before stopping the monitor.
+      boolean hasFatalDataValidationError = hasFatalDataValidationError(pushMonitor, resourceName);
```
Comment (mynameborat) [inline services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java:3268]: Looks like the resources on which the check is made is different across two code path? Is that intended?
The description only calls about moving out the place where this check happens but not necessarily changing the resource itself. e.g., Old one uses `versionTopicName` while this one uses `resourceName`. 

Are both same?

Comment (lluwm) [reply services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java:3268]: Yeah, good point and I also noticed that. Thus, I took a look at `deleteVersionFromStoreRepository` then realized that returned `deletedVersion` is Optional whose version number is either `versionNumber` or `empty`. 

Then, if consider that when we used the returned boolean `hasFatalDataValidationError` , it already passed the condition `deletedVersion.isPresent()`, then both value have to be the same.

**Comment by lluwm** [comment]
Thanks for the review, @majisourav99 and @mynameborat! Let me know if there is still comments on this PR. Otherwise, I will proceed to commit it.

**Pull request merged**